Lib/test/test_random.py

```diff
@@ -88,6 +89,25 @@ def test_shuffle(self):
         self.assertTrue(lst != shuffled_lst)
         shuffle(lst)
         self.assertTrue(lst != shuffled_lst)
+        self.assertRaises(TypeError, shuffle, (1, 2, 3))
+        self.assertRaises(TypeError, shuffle, {1, 2, 3})
+        self.assertRaises(TypeError, shuffle, 'shuffle')
+        self.assertRaises(KeyError, shuffle, dict(a=1, b=2))
```

Comment:
Let's only keep the first of these four tests.  It is reasonable to check to see if an immutable argument raises a TypeError.  The rest are either repeats of the same concept or an over-specification (what the current code happens to do rather than what is guaranteed).